### PR TITLE
docs: add __all__ variables to some modules

### LINKED ISSUE #638: [Python, Beginner] Add `__all__` attributes to modules.
The `__all__` variable is used to show which attributes of a module are public.
Please list all of the attributes that are imported in `icalendar/__init__.py` in their modules with the `__all__ = []` module variable.

See also:

- [`icalendar/__init__.py`](https://github.com/collective/icalendar/blob/main/src/icalendar/__init__.py)
- [What does `__all__` mean in Python?](https://stackoverflow.com/questions/44834/what-does-all-mean-in-python)

### PR DESCRIPTION
**Pull request by c0d33ngr**
This PR contributes to #638 

----
📚 Documentation preview 📚: https://icalendar--690.org.readthedocs.build/

### CODE REVIEW

**Comment by niccokunzmann** [comment]
@devdanzin could you have a look? If you see anything you would like to improve, please create a pull request to this one or suggest edits (I do not know how to do the last one) It could also be merged and another one created...

@c0d33ngr The tests are failing but I think that this is not your fault. I reported an issue: #677 Thanks for contributing so we can catch that problem with out CI setup!

Could you create a changelog entry for the next release?

File: src/icalendar/cal.py [hunk 1]
```diff
@@ -764,3 +764,8 @@ def example(cls, name) -> Calendar:
 # These are read only singleton, so one instance is enough for the module
 types_factory = TypesFactory()
 component_factory = ComponentFactory()
+
+__all__ = ["Alarm", "Calendar", "Component", "ComponentFactory", "Event",
+           "FreeBusy", "INLINE", "Journal", "Timezone", "TimezoneDaylight",
+           "TimezoneStandard", "Todo", "component_factory", "dateutil",
```
Comment (devdanzin) [inline src/icalendar/cal.py]:
```suggestion
           "TimezoneStandard", "Todo", "component_factory", "types_factory"]
```

File: src/icalendar/cal.py [hunk 2]
```diff
@@ -764,3 +764,8 @@ def example(cls, name) -> Calendar:
...
+           "types_factory"]
```
Comment (devdanzin) [inline src/icalendar/cal.py]:
```suggestion
```

**Comment by devdanzin** [comment]
> @devdanzin could you have a look? If you see anything you would like to improve, please create a pull request to this one or suggest edits (I do not know how to do the last one)

Sure, I'll check the names against imports and any comments regarding whether they are public. I've suggest an edit to remove dateutil, I think @c0d33ngr only has to accept the suggested changes and they will be automatically merged.

**Comment by devdanzin** [comment]
Not from this PR, but in `__init__.py` there seems to be a typo ("vTypesFactory" -> "TypesFactory"):
https://github.com/collective/icalendar/blob/7fb15675916914bc0057cc5b04bb508f10671b99/src/icalendar/__init__.py#L52

`get_example()` seems to be missing from `__all__` in cal.py, is it part of the public interface?

`ICAL_TYPE` seems to be missing from  `__all__` in parser_tools.py.

@c0d33ngr, would you like me to add `__all__` to prop.py?

**Comment by c0d33ngr** [comment]
> Not from this PR, but in `__init__.py` there seems to be a typo ("vTypesFactory" -> "TypesFactory"):
> 
> https://github.com/collective/icalendar/blob/7fb15675916914bc0057cc5b04bb508f10671b99/src/icalendar/__init__.py#L52
> 
> `get_example()` seems to be missing from `__all__` in cal.py, is it part of the public interface?
> 
> `ICAL_TYPE` seems to be missing from `__all__` in parser_tools.py.
> 
> @c0d33ngr, would you like me to add `__all__` to prop.py?

Yeah... Please do for the prop.py

**Comment by niccokunzmann** [comment]
@c0d33ngr and @devdanzin Thanks for collaborating on this PR. We are watching the test runs because I reported that they fail and I do not know what to do: https://github.com/coverallsapp/github-action/issues/217

So, thanks for working as contributors so that people in the future do not get broken test results.

**Comment by devdanzin** [comment]
> @c0d33ngr and @devdanzin Thanks for collaborating on this PR. We are watching the test runs because I reported that they fail and I do not know what to do: [coverallsapp/github-action#217](https://github.com/coverallsapp/github-action/issues/217)

If you'd like me to push dummy commits (in a new PR, I guess) so coveralls runs and you can get more data points, let me know.

**Comment by niccokunzmann** [comment]
Thanks for the help. I think, any work you do together here will create a new commit and then, we know more. There is no urgency and you do not have to consider that in a special way. I just wanted to let you know :) If you like to take up another issue though, we might find that other commits you make might show the same symptoms ;)

**Comment by c0d33ngr** [comment]
> > @devdanzin could you have a look? If you see anything you would like to improve, please create a pull request to this one or suggest edits (I do not know how to do the last one)
> 
> Sure, I'll check the names against imports and any comments regarding whether they are public. I've suggest an edit to remove dateutil, I think @c0d33ngr only has to accept the suggested changes and they will be automatically merged.

Thank you for the code review you did 😊

**Comment by devdanzin** [comment]
Thank you for being so receptive!

Unfortunately I cannot push to this PR, so I've created a PR against it: https://github.com/c0d33ngr/icalendar/pull/1.

The same changes that go into that one go to https://github.com/collective/icalendar/pull/693 to see if I can trigger coveralls failures, but no luck so far :(

**Comment by niccokunzmann** [comment]
@c0d33ngr, have a look here: https://github.com/c0d33ngr/icalendar/pull/1 there are some proposed changes.

**Comment by c0d33ngr** [comment]
> @c0d33ngr, have a look here: [c0d33ngr#1](https://github.com/c0d33ngr/icalendar/pull/1) there are some proposed changes.

I merged @devdanzin branch into mine

**Comment by niccokunzmann** [comment]
Thanks!

**Pull request merged**